# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/android/SessionAccessibility.h
# Commit: 4965ca9b7882
# Full Hash: 4965ca9b7882e482c0d5f48de63fad4457a9863f
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-07-07 09:37:57
# Regressor Bug: 1771934
# File Overlap Count: 1
# Description:
#   Bug 1771934 - Unregister all SessionAccessibility accessibles when top doc is shut down. r=Jamie
#   
#   The associated PresShell of the root doc can be used for retrieving the
#   SessionAccessibility. If the PresShell is about to go away, we should
#   unregister all the accessibles.
# ==============================================================================

diff -r 3827c37b916d -r 4965ca9b7882 accessible/android/SessionAccessibility.h
--- a/accessible/android/SessionAccessibility.h	Thu Jul 07 01:47:08 2022 +0300
+++ b/accessible/android/SessionAccessibility.h	Wed Jul 06 22:26:08 2022 +0000
@@ -46,6 +46,7 @@
 
   static void Init();
   static RefPtr<SessionAccessibility> GetInstanceFor(Accessible* aAccessible);
+  static RefPtr<SessionAccessibility> GetInstanceFor(PresShell* aPresShell);
 
   // Native implementations
   using Base::AttachNative;
@@ -113,6 +114,7 @@
 
   static void RegisterAccessible(Accessible* aAccessible);
   static void UnregisterAccessible(Accessible* aAccessible);
+  static void UnregisterAll(PresShell* aPresShell);
 
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(SessionAccessibility)
 
